manifests/redis_exporter.pp

```diff
@@ -0,0 +1,181 @@
+#
+#  [*init_style*]
+#  Service startup scripts style (e.g. rc, upstart or systemd)
+#
+#  [*install_method*]
+#  Installation method: url or package (only url is supported currently)
+#
+#  [*manage_group*]
+#  Whether to create a group for or rely on external code for that
+#
+#  [*manage_service*]
+#  Should puppet manage the service? (default true)
+#
+#  [*manage_user*]
+#  Whether to create user or rely on external code for that
+#
+#  [*namespace*]
+#  Namespace for the metrics, defaults to `redis`.
+#
+#  [*os*]
+#  Operating system (linux is the only one supported)
+#
+#  [*package_ensure*]
+#  If package, then use this for package ensure default 'latest'
+#
+#  [*package_name*]
+#  The binary package name - not available yet
+#
+#  [*purge_config_dir*]
+#  Purge config files no longer generated by Puppet
+#
+#  [*restart_on_change*]
+#  Should puppet restart the service on configuration change? (default true)
+#
+#  [*service_enable*]
+#  Whether to enable the service from puppet (default true)
+#
+#  [*service_ensure*]
+#  State ensured for the service (default 'running')
+#
+#  [*service_name*]
+#  Name of the node exporter service (default 'redis_exporter')
+#
+#  [*user*]
+#  User which runs the service
+#
+#  [*version*]
+#  The binary release version
+class prometheus::redis_exporter (
+  $addr                 = $::prometheus::params::redis_exporter_redis_address,
+  $arch                 = $::prometheus::params::arch,
+  $bin_dir              = $::prometheus::params::bin_dir,
+  $download_extension   = $::prometheus::params::redis_exporter_download_extension,
+  $download_url         = undef,
+  $download_url_base    = $::prometheus::params::redis_exporter_download_url_base,
+  $extra_groups         = $::prometheus::params::redis_exporter_extra_groups,
+  $extra_options        = '',
+  $group                = $::prometheus::params::redis_exporter_group,
+  $init_style           = $::prometheus::params::init_style,
+  $install_method       = $::prometheus::params::install_method,
+  $manage_group         = true,
+  $manage_service       = true,
+  $manage_user          = true,
+  $namespace            = 'redis',
+  $os                   = $::prometheus::params::os,
+  $package_ensure       = $::prometheus::params::redis_exporter_package_ensure,
+  $package_name         = $::prometheus::params::redis_exporter_package_name,
+  $purge_config_dir     = true,
+  $restart_on_change    = true,
+  $service_enable       = true,
+  $service_ensure       = 'running',
+  $service_name         = 'redis_exporter',
+  $user                 = $::prometheus::params::redis_exporter_user,
+  $version              = $::prometheus::params::redis_exporter_version,
+) inherits prometheus::params {
+
+  $release = "v${version}"
+
+  $real_download_url = pick($download_url, "${download_url_base}/download/${release}/${package_name}-${release}.${os}-${arch}.${download_extension}")
+  validate_bool($purge_config_dir)
```

Comment:
you can drop these validate_* calls, they're deprecated and replaced by datatypes